### DISCUSSION TITLE
Docker build with node-alpine image

### DISCUSSION THREAD

**Discussion by sambeetkumarsahu** [mckaywrigley/chatbot-ui#1355] [Help]
Hi everyone can you please help me in building the app using the node-alpine image

**Comment by Jonneal3** [comment]
I already have one built.. I'll try and create a pull request or just post here soon

**Reply by sambeetkumarsahu** [reply]
Hi @Jonneal3 thanks

**Reply by ackvf** [reply]
#me2

**Reply by sambeetkumarsahu** [reply]
I Got is 

FROM node:20-slim AS base


FROM base AS deps
WORKDIR /app


COPY package.json package-lock.json ./
RUN npm install


FROM base AS builder
WORKDIR /app
COPY --from=deps /app/node_modules ./node_modules
COPY . .


ENV NEXT_TELEMETRY_DISABLED 1

RUN npm run build


FROM base AS runner
WORKDIR /app

ENV NODE_ENV production

ENV NEXT_TELEMETRY_DISABLED 1

RUN addgroup --system --gid 1001 nodejs
RUN adduser --system --uid 1001 nextjs

COPY --from=builder /app/public ./public


RUN mkdir .next
RUN chown nextjs:nodejs .next


COPY --from=builder --chown=nextjs:nodejs /app/.next/standalone ./
COPY --from=builder --chown=nextjs:nodejs /app/.next/static ./.next/static

USER 1001

EXPOSE 3000

ENV PORT 3000

ENV HOSTNAME "0.0.0.0"


CMD ["node", "server.js"]


@ackvf

**Reply by ackvf** [reply]
thanks @sambeetkumarsahu , btw, you can use code block like this using ` ``` codeblock ``` `

e.g.
``` 
    ```
    FROM node:20-slim AS base

    # Install dependencies only when needed
    FROM base AS deps
    WORKDIR /app
    ```
```

BTW2, right now I am using this https://github.com/mckaywrigley/chatbot-ui/discussions/1379, it comes with a single docker image and just runs without any configuration. I think it might be version 1 as it uses localstorage and no postgress db, 1) but I don't mind really as long as it works 2) I actually kinda prefer this version, because I can easily edit the chat history.

**Comment by spammenotinoz** [comment]
Suggest using an LTS as that will be closer to the AWS Vercel deployment.
For a laugh, I just knocked up a docker build (front-end only), passing the env variables as build arguments, but you need to make sure you build is private.
File Uploads are working, but other than that seems okay.  I run on Vercel so have no real need for docker, just did as a test only.
